Beta v4 Release

verndrade opened the pull request:
- New login system (Professors can Log In!)

- Wait for permission before showing syllabi

- UI/UX and Accessibility Improvements

kyawzazaw committed:
Update README

Use bulletpoints instead of long paragraphs for easier reading.
Rephrase sentences to be more succinct.

Add a roadmap section for features intended to be developed in the future.

kyawzazaw committed:
Update README

Additional formatting

winstliu committed:
Add number to Submission and wire it up

winstliu committed:
Automatically migrate existing submissions

winstliu committed:
:fire: course and make number non-null

winstliu committed:
Merge branch 'development' into wl-split-course

verndrade committed:
hide all submission on entry

winstliu committed:
Merge branch 'development' into wl-split-course

winstliu committed:
Vernon this might help your feedback

winstliu committed:
Don't forget the migration

winstliu committed:
Rename some fields on School to make more sense

poster->creator
school->name

winstliu committed:
Merge branch 'development' into wl-split-course

winstliu committed:
Migration conflicts -,-

winstliu committed:
Link Submissions to a School

winstliu committed:
Redirect to upload instead of rendering

The POSTs were going to the wrong place.

winstliu committed:
Link Submission to User

winstliu committed:
Create UserProfile

Link UserProfile -> User and UserProfile -> School

winstliu committed:
And thus, no more need for get_domain

winstliu committed:
Use our new relations to define the leaderboard :)

winstliu committed:
Remove uploads JSONField

winstliu committed:
Create school on UserProfile creation if necessary

winstliu committed:
Don't forget the uploads migration

winstliu committed:
Signals suck; hook into the social auth pipeline instead

winstliu committed:
Revert "Signals suck; hook into the social auth pipeline instead"

This reverts commit da542b424f7ea8d54182e997e6e05deca8377b66.

winstliu committed:
Go back to signals to facilitate test-writing

(The tests are broken though b/c I'm not handling non-edu emails)

winstliu committed:
Make profile.school optional

This allows the email logic to work again, and fixes the tests

verndrade committed:
Merge pull request #66 from SyllabiShare/wl-update-feedback-link

Vernon this might help your feedback

verndrade committed:
Delete 0015_submission_hidden.py

verndrade committed:
stuff

verndrade committed:
add new model field

verndrade committed:
fix uploads

verndrade committed:
admin changes

verndrade committed:
toggle hidden on admin

winstliu committed:
Merge branch 'development' into wl-split-course

winstliu committed:
Merge branch 'development' into wl-redirect-to-upload

winstliu committed:
Keep the form generic

winstliu committed:
Merge branch 'development' into wl-link-models

winstliu committed:
Migration magic woo -,-

verndrade committed:
Update settings.py

verndrade committed:
Merge pull request #67 from SyllabiShare/wl-redirect-to-upload

Redirect to upload instead of rendering

verndrade committed:
update upload page

verndrade committed:
Merge branch 'development' into wl-split-course

verndrade committed:
Merge pull request #64 from SyllabiShare/wl-split-course

Split course into dept and number

verndrade committed:
remove donation request

verndrade committed:
edit about page and change github links

verndrade committed:
Merge branch 'development' of https://github.com/verndrade/syllabi-share into vern-about-page

verndrade committed:
minor edits

verndrade committed:
revert unnecessary base changes

verndrade committed:
hide leaderboard

verndrade committed:
fix whitespace in dept field in uploads

winstliu committed:
Merge branch 'development' into wl-link-models

verndrade committed:
Merge pull request #69 from SyllabiShare/vern-about-page

About page changes

winstliu committed:
Yolo?

winstliu committed:
Merge pull request #68 from SyllabiShare/wl-link-models

Link models

winstliu committed:
Make the file input full-width

verndrade committed:
remove google auth, add verify email auth

verndrade committed:
Merge pull request #70 from SyllabiShare/wl-uncramp-file-input

Make the file input full-width

verndrade committed:
Merge pull request #53 from kyawzazaw/development

Update README with succint text and better formatting

winstliu committed:
Merge branch 'development' into alternate-login

# Conflicts:
#	templates/error.html

winstliu committed:
Merge Profile & UserProfile; try to merge migrations

winstliu committed:
crispy_forms first pass; very WIP

winstliu committed:
Enforce .edu emails

winstliu committed:
Nothing like regexes to make your day

verndrade committed:
fix clown mistake

🤡

winstliu committed:
Use Crispy helpers

winstliu committed:
Similar treatment for login

winstliu committed:
Split up the login/signup button

winstliu committed:
Use {% url %} and remove custom <a> styling

winstliu committed:
Remove special-casing of non-edu emails

That is handled within SignupForm now

winstliu committed:
Migration!

winstliu committed:
Haha! Pretend to use email as the primary authentication ID

mengsarah committed:
more responsive number/year and added January

winstliu committed:
Initial UI revamp of the landing page

mengsarah committed:
rearranged hardcoded messages and fixed January

mengsarah committed:
fixed typo

mengsarah committed:
Revert "fixed typo"

This reverts commit dd0521cbcd46f616c7932ffe09122ccf37400099.

mengsarah committed:
Revert "Revert "fixed typo""

This reverts commit 05a8f5a29ad7369d2f3b41c6e00e0716d3e96bdf.

mengsarah committed:
added success and warning classes

mengsarah committed:
gave messages success/warning classes + converted the on-page-load success and error messages into Django style

mengsarah committed:
adding myself

mengsarah committed:
downsized picture oops

mengsarah committed:
turns out messages makes HTML into plaintext

verndrade committed:
about page changes

verndrade committed:
fix space

winstliu committed:
Switch the landing form to signup

Rationale: you'll be redirected to the index page when you're logged in,
so it's more likely that you'll see the landing page
when you don't have an account.

winstliu committed:
Add sign up/sign in to the nav bar

winstliu committed:
Transition signup/sign in to the new template

winstliu committed:
I hate crispy

winstliu committed:
Transition about & privacy

winstliu committed:
Remove custom link styling

winstliu committed:
Hohoho figure out how to programmatically adjust the navbar

winstliu committed:
No need for six

mengsarah committed:
Merge pull request #72 from SyllabiShare/UI-improvement (upload success/error messages)

UI improvement: Upload success/error messages

winstliu committed:
Switch school.html to use landing

winstliu committed:
landing -> base, base -> base-deprecated

winstliu committed:
Merge branch 'development' into alternate-login

# Conflicts:
#	syllabiShare/views.py
#	templates/about.html

winstliu committed:
I can't really read

winstliu committed:
Update sorry.html

winstliu committed:
error.html -> landing.html

winstliu committed:
Center the introductory text

winstliu committed:
Accessibility adjustments

winstliu committed:
Show a static page for confirmation

winstliu committed:
Prevent duplicate email signup

winstliu committed:
Show error when logging in as unconfirmed user

winstliu committed:
Add a TODO for activation failure

winstliu committed:
Let's not hardcode the email...

winstliu committed:
Password reset

winstliu committed:
Wordsmith the unconfirmed email message

winstliu committed:
And...we're using "sign in", not "login"

winstliu committed:
🎨

winstliu committed:
Move stuff around so we can use defaults

mengsarah committed:
first try at AA color and link contrast compliance

verndrade committed:
fix re-add on refresh

winstliu committed:
Merge branch 'development' into alternate-login

# Conflicts:
#	syllabiShare/views.py

verndrade committed:
small ui changes

verndrade committed:
Merge pull request #71 from SyllabiShare/alternate-login

Better auth

verndrade committed:
Merge branch 'master' of https://github.com/verndrade/syllabi-share into development

verndrade committed:
delete account hotfix

mengsarah committed:
minor typographical fixes/improvements

mengsarah committed:
added helpful notes

mengsarah committed:
added alt text (feeling conflicted about alt text for headshots)

mengsarah committed:
first try at better CSS for a11y: colors and focus

mengsarah committed:
update nav: +"browse", changed breakpoint from md to lg

mengsarah committed:
making use of the C in CSS

winstliu committed:
Redo parts of 243ae63

Signup/login will get different UI in a later commit

winstliu committed:
Redo auth styles

- Signup/login get styles from the forgot password flow

- Signup has a max width of 600px; all others, 400px

- Removed unnecessary level of nesting

winstliu committed:
Use decorators to control auth flow

winstliu committed:
Unrelated changes

mengsarah committed:
let's check for messages on every page! Probably won't break anything. Probably.

mengsarah committed:
text-light class not needed with this branch's style.css

verndrade committed:
Merge pull request #76 from SyllabiShare/wl-auth-decorators

Use decorators to control auth flow

verndrade committed:
Merge pull request #74 from SyllabiShare/wl-account-creation-ui-followup

Account creation: UI followup

verndrade committed:
Update README.md

winstliu committed:
We don't need personal info

winstliu committed:
Link to signup page on the login page

mengsarah committed:
merge with forgotten some-accessibility branch

winstliu committed:
Provide a way to resend confirmation emails

verndrade committed:
Merge pull request #79 from SyllabiShare/wl-rm-name

We don't need personal info

mengsarah committed:
beginnings of breadcrumbs

mengsarah committed:
better messages for user + slight if/else refactor

mengsarah committed:
Getting rid of that back button :fire:

mengsarah committed:
fine-tuned breadcrumbs

mengsarah committed:
fixed that SyllabiShare linkedin link, removed empty link

winstliu committed:
Merge branch 'development' into wl-resend-confirmation

winstliu committed:
Rudimentary email spam protection

winstliu committed:
Merge remote-tracking branch 'origin/wl-resend-confirmation' into wl-resend-confirmation

winstliu committed:
🎨

verndrade committed:
Merge pull request #81 from SyllabiShare/wl-resend-confirmation

Provide a way to resend confirmation emails

verndrade committed:
Merge pull request #80 from SyllabiShare/wl-link-to-login--in-signup

Link to signup page on the login page

mengsarah committed:
display signup page errors all at the top, too

mengsarah committed:
you don't get to sign up if you're already logged in

mengsarah committed:
right, signup isn't the only place that throws actual form errors

mengsarah committed:
Merge branch 'development' into UI-improvement

mengsarah committed:
updated due to grabbing updates from the development branch

mengsarah committed:
image descriptions

winstliu reviewed (approved):
Rubber stamp approval

When are we switching Heroku to master instead of development, again?

verndrade commented:
I'm waiting on the accessibility and bookmark PR. once we merge to master ill switch the deployed branch to master. need to get that pipeline thing sometime, but if not we can use syllabi-share-staging to deploy development and I can pg:copy prod data every now and then

mengsarah committed:
DOWN WITH THAT BACK BUTTON

mengsarah committed:
after much wrangling, goodbye fixed GitHub icon :fire: and hellooo footer! + minor misc

footer can be filled with more stuff later; misc: removed lingering back button, switched background image to just the plain color it is, added some header-like comments to CSS now that it's getting big

mengsarah committed:
considering I already changed the search button color, guess it can't hurt to make it AAA-compliant

my eyes: sarah can you please make the darn thing AAA-compliant because there's something about this particular combination of AA-compliant dark gray and green that just isn't cutting it for us... pretty please with a cherry on top

mengsarah committed:
gave search a role

mengsarah committed:
moving JS-delivered error message to next row to not cut off the text for file upload

mengsarah committed:
more compliant colors (text-muted, primary)

verndrade committed:
move breadcrumbs

verndrade committed:
more breadcrumbs

verndrade committed:
Merge pull request #77 from SyllabiShare/UI-improvement

Improving UI/UX and accessibility
